tests/pipeline/utils.py

```diff
@@ -3,7 +3,8 @@
 import random
 from os import environ
 import io
-
+import os
```

Comment:
this is indeed a file that deserve the name `utils`.
It also twists my brain to look at it. I wouldn't count on me detecting any bugs in your changes here

Reply:
OK :)